app/controllers/application_controller.rb

```diff
@@ -21,4 +29,10 @@ def nav_presenter
 
     @nav_presenter ||= NavPresenter.new(params: params, current_user: current_user)
   end
+
+  protected
+
+  def authenticate_user!
+    auth_method?('form') ? :authenticate_user_via_devise! : authenticate_user_via_oauth!
```

Comment:
```suggestion
    auth_method?('form') ? authenticate_user_via_devise! : authenticate_user_via_oauth!
```

Reply:
basically spec should not rely on the setting in .env file, so I must fix it

Reply:
fixed, there was a problem with request specs related to my last changed. Although maybe in case of auth spec it's justified to have it, but for other stuff I think it's kinda redundant to add request specs if there are respective feature specs.